test/unit/actions/run_hosts_job_test.rb

```diff
@@ -39,8 +39,7 @@ class RunHostsJobTest < ActiveSupport::TestCase
     end
 
     it 'triggers the RunHostJob actions on the resolved hosts in run phase' do
-      template_invocation = job_invocation.template_invocation_for_host(host)
-      action.expects(:trigger).with(Actions::RemoteExecution::RunHostJob, job_invocation, host, template_invocation, proxy)
+      action.expects(:trigger).with() { |*args| args[0] == Actions::RemoteExecution::RunHostJob }
```

Comment:
hard-coding of API makes it fragile, also the template_invocation is now built (not saved) as part of `RunHostsJob` run phase so can't be found here